trestle/core/rules_validator.py

```diff
@@ -0,0 +1,145 @@
+                                    f'control: {imp_req.control_id} in component: {component.title} is not consistent '
+                                    'with other values provided across controls. Invalid model'
+                                )
+                                return False
+                            shared_params[shared_param_id].append(
+                                {
+                                    'level': 'ip', 'control_id': imp_req.control_id, 'values': shared_param['values']
+                                }
+                            )
+                        continue
+                    for set_param in as_list(imp_req.set_parameters):
+                        # if not a rule parameter then ignore
+                        if set_param.param_id not in (control_imp_rule_param_ids + imp_req_rule_param_ids):
+                            continue
+                        # gets existing param if exists
+                        existing_params = [x for x in shared_params[set_param.param_id] if x['level'] != 'ci']
+                        if not existing_params:
+                            # adds a non exisitng param across controls for revision
+                            shared_params[set_param.param_id].append(
+                                {
+                                    'level': 'ip', 'control_id': imp_req.control_id, 'values': set_param.values
+                                }
+                            )
+                            opts_for_param = [
+                                p['options'] for p in shared_params[set_param.param_id] if p['level'] == 'ci'
+                            ][0]
+                            # validates if param value exist inside value alternatives for the param
+                            for value in set_param.values:
+                                if value not in opts_for_param[0]:
+                                    logger.warning(
+                                        f'Rule parameter value: {value} for parameter: {set_param.param_id} in '
+                                        f'control {imp_req.control_id} for component: {component.title} '
+                                        f'is not in the alternatives provided by default: {opts_for_param[0]} '
+                                        'for the control implementation'
+                                    )
+                            continue
+                        diff = self.val_diff_param_values(existing_params, set_param.values)
+                        if diff:
+                            logger.error(
+                                f'Rule parameter value: {diff} provided for: {set_param.param_id} in '
+                                f'control: {imp_req.control_id} in component: {component.title} is not consistent with '
+                                'other values provided across controls. Invalid model'
+                            )
```

Comment:
Please check the other validators to see how they indicate VALID or INVALID.  We should keep the messaging structure the same.